src/lib/dialog/dialog-ref.ts

```diff
@@ -14,10 +14,18 @@ export class MdDialogRef<T> {
   /** The instance of component opened into the dialog. */
   componentInstance: T;
 
+  /** Expose overlay backdrop click event */
+  backdropClicked: Observable<void>;
+
+  /** Subject for notifying the user that esc key way pressed */
+  escapePressed = new Subject<void>();
```

Comment:
Check https://github.com/angular/material2/pull/2855

Imagine you keep state of your modal in a global store, like redux or @ngrx/store, so you create component that looks like this.

```html
<my-dialog [open]="modalState$ | async">
  <button (click)="doSomethingThatChangesModalState$toFalse()">close</button>
</my-dialog>
```

Now you are in full control of the state, when you change `modalState$`, your modal closes, BUT when you click on backdrop or press escape, your modal closes while your `modalState$` is still true and your store is out of sync. You could catch up by changing `modalState$` to `false` on `afterClose()`, but that could for example be called only after the animation finishes, it simply smells, it's the other way around than it should be. So you want to disable the default closing behaviour of `backdropClick` and `escape`, so you do `disableClose: true`, but then you have to manually set up listeners for both, which will be big pain in the ass.


So ideally you want to do

```html
<my-dialog 
    [open]="modalState$ | async" 
    (close)="doSomethingThatChangesModalState$toFalse()">
  <button (click)="doSomethingThatChangesModalState$toFalse()">close</button>
</my-dialog>
```

where close event happens on either `backdropClick` or `escape` press https://github.com/fxck/material2/blob/662703b63831797c9969dda4dd03d69d794aa1f4/src/lib/dialog/dialog-element.ts#L54-L61